getnacelle/nacelle-js
# feat: implement the `productCollections` method

## Why are these changes introduced?

Addresses [ENG-8325](https://nacelle.atlassian.net/browse/ENG-8325) 

> Implement the `productCollections` method



## What is this pull request doing?



1. Adds the `productCollections` method to `@nacelle/storefront-sdk-plugin-commerce-queries` to match `@nacelle/storefront-sdk@1.x`'s `productCollections` method.
2. Refactors some `console` side effects in other methods with new utilities.
3. Refactors some tests to simplify and improve clarity.

## How to Test



1. Check out the `ENG-8325-implement-the-productcollections-method` branch.
2. Navigate to `packages/storefront-sdk-plugins/commerce-queries`.
4. `npm run coverage` - all tests should pass with 100% coverage.

![ENG-8325-implement-the-productcollections-method branch all test passing with 100% coverage](https://user-images.githubusercontent.com/5732000/215841130-044fdb07-1d98-45b7-b5e5-c9c4a47375c5.png)

5. `npm run build` - there should be no build errors.

From there, you can test in a local project via `npm link`. In my local testing, I found that the `productCollections` method works as expected. For example, the following request returned the expected number of collections & products in the collection:

```ts
const { data } = await client.productCollections({
  maxReturnedEntries: 1,
  maxReturnedEntriesPerCollection: 2,
});
```

![@nacelle:storefront-sdk-plugin-commerce-queries demo productCollections](https://user-images.githubusercontent.com/5732000/215908050-9aa9b497-7cae-494b-9ac6-656229ac0298.png)


## Checklist



[ENG-8325]: https://nacelle.atlassian.net/browse/ENG-8325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
